share/rpcauth/rpcauth.py

```diff
@@ -9,26 +9,36 @@
 import base64
 import hmac
 
-if len(sys.argv) < 2:
-    sys.stderr.write('Please include username as an argument.\n')
-    sys.exit(0)
+def generate_salt():
+    # This uses os.urandom() underneath
+    cryptogen = SystemRandom()
 
-username = sys.argv[1]
+    # Create 16 byte hex salt
+    salt_sequence = [cryptogen.randrange(256) for i in range(16)]
+    return ''.join([format(r, 'x') for r in salt_sequence])
 
-#This uses os.urandom() underneath
-cryptogen = SystemRandom()
+def generate_password(salt):
+    # Create 32 byte b64 password
```

Comment:
nit: prefer docstrings for function-level comments.

Reply:
Done